core/src/main/kotlin/org/evomaster/core/output/naming/RestActionTestCaseNamingStrategy.kt

```diff
@@ -37,6 +43,66 @@ open class RestActionTestCaseNamingStrategy(
         }
     }
 
+    /**
+     * In REST Individuals. Ambiguity will be resolved with the path filter.
+     * UriParams and QueryParams filters will be left for experimentalPurposes.
+     *
+     * Whenever an ambiguity is solved, then it should remove that test from the cycle. There is no need to execute the
+     * following filters
+     */
+    override fun resolveAmbiguity(individualToName: MutableMap<EvaluatedIndividual<*>, String>, duplicatedIndividuals: MutableSet<EvaluatedIndividual<*>>) {
+        checkForPath(individualToName, duplicatedIndividuals)
+    }
+
+    /*
+     * If any test has a different path, then add previous segment to them to make them differ.
+     * The filter call ensures that we are only performing this disambiguation when there's only one individual that
+     * differs and when said individual does not have a parameter as a last element since it might differ in the
+     * parameter name but not the rest of the path.
+     */
+    private fun checkForPath(individualToName: MutableMap<EvaluatedIndividual<*>, String>, duplicatedIndividuals: MutableSet<EvaluatedIndividual<*>>) {
+        val groupByPath = duplicatedIndividuals.groupBy {
+            var path = (it.evaluatedMainActions().last().action as RestCallAction).path
+            if (path.isLastElementAParameter()) {
+                path = path.parentPath()
+            }
+            val toStringPath = path.toString()
+            val isLastAParam = path.isLastElementAParameter()
+            Pair(toStringPath, isLastAParam)
+        }.filter { it.value.size == 1 && !it.key.second }
+
+        groupByPath.forEach { entry ->
```

Comment:
why introducing local variable `groupByPath` and not chain the `forEach` directly after the filter?

Reply:
To be honest, readability and debugging purposes at the time